python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -664,8 +664,15 @@ def get_nodes_for(node_types: Dict[NodeType, NodeTypeConfigDict],
 
 
 def _utilization_score(node_resources: ResourceDict,
-                       resources: ResourceDict) -> float:
+                       resources: List[ResourceDict]) -> float:
     remaining = copy.deepcopy(node_resources)
+    is_gpu_node = "GPU" in node_resources
+    any_gpu_task = any("GPU" in r for r in resources)
+
+    # Avoid launching GPU nodes if there aren't any GPU tasks at all. Note that
+    # if there *is* a GPU task, then CPU tasks can be scheduled as well.
+    if is_gpu_node and not any_gpu_task:
```

Comment:
On issue here is that if all the available nodes are gpu nodes you will never scale up if you have CPU tasks only and instead you will print to the user the "The autoscaler could not find a node type to satisfy the ..." message which is bad. 
Please fix that before merging this PR.

Reply:
I thought about that, and think in this case this is the right behavior. The user should be adding some CPU nodes to their config, rather than Ray auto launching expensive GPU nodes.